pkg/network/infraconfigurators/bridge.go

```diff
@@ -105,11 +131,12 @@ func (b *BridgePodNetworkConfigurator) PreparePodNetworkInterface() error {
 
 	if b.ipamEnabled {
 		// Remove IP from POD interface
-		err := b.handler.AddrDel(b.podNicLink, &b.podIfaceIP)
-
-		if err != nil {
-			log.Log.Reason(err).Errorf("failed to delete address for interface: %s", b.podNicLink.Attrs().Name)
-			return err
+		if b.podIfaceIP != nil {
```

Comment:
isn't this redundant ? if you have IPAM enabled (the criteria to have IPAM enabled is to have an IP in the pod interface), you'll have a pod interface IP.

Reply:
`b.podIfaceIP` is the IPv4 address, it might still be `nil` for IPv6-only pods.